src/components/FavoriteButton/FavoriteButton.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useContext } from 'react';
+import classNames from 'classnames';
+import './FavoriteButton.scss';
+
+import { Product } from '../../types/Product';
+import { ProductsContext } from '../../helpers/ProductsContext';
+import favoritesIcon from '../../images/favorites.svg';
+import favoritesIconRed from '../../images/favorites-red.svg';
+
+type Props = {
+  product: Product;
+  isBigButton?: boolean;
+};
+
+export const FavoriteButton: React.FC<Props> = ({ product, isBigButton }) => {
+  const {
+    addProductToFavorites,
+    isInFavorites,
+  } = useContext(ProductsContext);
+
+  const handleClick = () => addProductToFavorites(product);
+
```

Comment:
const imageSrc = useMemo(()
```suggestion
const imageSrc = useMemo(() => {
 const isFavoriteProduct = isInFavorites(product);
 
 return isFavoriteProduct ? favoritesIconRed : favoritesIcon;
}, [....])
```